### PR TITLE
Realtime push

### PR DESCRIPTION
**Pull request by radiofreemattd**
This is an initial attempt to provide real-time notification to the board. This uses Pusher to force a `window.reload` when a piece is moved. Currently, both players screens refresh, which is less than ideal, but for an MVP, this seems fine to me.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -44,6 +44,12 @@ gem 'omniauth-facebook'
 # Use omniauth-google-oauth2 for google login
 gem 'omniauth-google-oauth2'
 
+# Use Pusher for realtime updates
+gem 'pusher'
+
+# Use Figaro for environment configuration
+gem 'figaro'
```
Comment (jonmast) [inline Gemfile]: I think we only need this in the development environment?

File: app/controllers/games_controller.rb [hunk 2]
```diff
@@ -21,6 +21,7 @@ def create
   end
 
   def show
+    @game = current_game
```
Comment (jonmast) [inline app/controllers/games_controller.rb]: This line could be removed and everything would still work. Maybe we should call current_game in the view for consistency?

File: app/controllers/pieces_controller.rb [hunk 3]
```diff
@@ -48,7 +48,9 @@ def message
   end
 
   def attempt_move
-    selected_piece.move_to!(destination_coordinates) if moving_validly?
+    return unless moving_validly?
+    selected_piece.move_to!(destination_coordinates)
+    Pusher.trigger(current_game.id, 'refresh_event', message: '') unless Rails.env.test?
```
Comment (jonmast) [inline app/controllers/pieces_controller.rb]: :+1: Looks good.

I forsee a potential situation where multiple developers are testing a game with the same id in the dev environement and refreshing each other's pages :smile:

File: app/views/layouts/application.html.slim [hunk 4]
```diff
@@ -2,18 +2,15 @@ doctype html
 html
   head
     title ChessApp
+    script src="https://js.pusher.com/3.0/pusher.min.js"
     = stylesheet_link_tag    "application", media: "all"
-    = javascript_include_tag    'application'
-    = stylesheet_link_tag    "application", media: "all", "data-turbolinks-track" => true
-    = javascript_include_tag "application", "data-turbolinks-track" => true
```
Comment (jonmast) [inline app/views/layouts/application.html.slim]: You should fire the guy that introduced this duplication :smiling_imp:

**Comment by jonmast** [comment]
Great work, :heart: it

File: app/views/layouts/application.html.slim (hunk 4 above)
Comment (mscoutermarsh) [inline app/views/layouts/application.html.slim]: lol :D

**Comment by mscoutermarsh** [comment]
I love the "mvp" approach here.

File: app/controllers/pieces_controller.rb (hunk 3 above)
Comment (mscoutermarsh) [inline app/controllers/pieces_controller.rb]: :smile: For this, could have 2 sets of credentials. One for local dev & one for production

**Pull request merged**